modules/gateway/controllers/service/populate/virtualservice.go

```diff
@@ -113,15 +113,22 @@ func newRoute(systemNamespace, externalGW string, published bool, portBinding v1
 		if dests[0].Subset != "" {
 			host = host + "-" + dests[0].Subset
 		}
+		port := portBinding.Port
+		if autoscale && deployIsZero(svc) {
+			host = "activator"
+			ns = systemNamespace
+			port = 8012
+		}
 		route.Headers = &v1alpha3.Headers{
 			Request: &v1alpha3.HeaderOperations{
 				Set: map[string]string{
-					"l5d-dst-override": fmt.Sprintf("%v.%s.svc.cluster.local:%v", host, ns, portBinding.Port),
+					"l5d-dst-override": fmt.Sprintf("%v.%s.svc.cluster.local:%v", host, ns, port),
```

Comment:
Should we turn `l5d-dst-override` into a const too ? That also brings up "l5d-remote-ip" and "l5d-server-id". I think it would be better to at least do the override, a const can avoid typos, also easier to search or change all uses.